lib/resty/acme/client.lua

```diff
@@ -498,31 +499,38 @@ function _M:order_certificate(domain_key, ...)
     for _, challenge in ipairs(challenges.challenges) do
       local typ = challenge.type
       if self.challenge_handlers[typ] then
-        local err = self.challenge_handlers[typ]:register_challenge(
-          challenge.token,
-          challenge.token .. "." .. self.account_thumbprint,
-          {...}
-        )
-        if err then
-          return nil, "error registering challenge: " .. err
-        end
-        registered_challenges[registered_challenge_count + 1] = challenge.token
-        registered_challenge_count = registered_challenge_count + 1
-        log(ngx_DEBUG, "register challenge ", typ, ": ", challenge.token)
-        -- signal server to start challenge check
-        -- needs to be empty json body rather than empty string
-        -- https://tools.ietf.org/html/rfc8555#section-7.5.1
-        local _, _, err = self:post(challenge.url, {})
-        if err then
-          return nil, "error start challenge check: " .. err
+        if challenge.status == 'pending' then
```

Comment:
i think we should swap the two `if` statement, since there could be a challenge that we didn't enable but is already valid:
```lua
if challenge.status == 'pending' then
  if self.challenge_handlers[typ] then
    -- register and signal
  end
else
  if challenge.status == 'valid' then
    has_valid_challenge = true
  end
  -- log
end
```

actually to reduce the indention, how about:
```lua
if challenge.status ~= 'pending' then
  if challenge.status == 'valid' then
    has_valid_challenge = true
  end
  -- log
elseif self.challenge_handlers[typ] then
  -- register and signal
end
```

Reply:
Good catch! Done